### DISCUSSION TITLE
Use Airflow stable API to get Run-id

### DISCUSSION THREAD

**Discussion by Ashay4u** [apache/airflow#21613] [Q&A]
Hi

I am trying to create a system through which I first list all the failed task instances of a dag and then later want to pick and chose which ones i want to clear (for re-run) using the stable airflow API. 
I get the list of failed task instances using a call like below

curl --user 'test:tes' -X GET 'http://airflow-url:port/api/v1/dags/example_bash_operator/dagRuns/?/taskInstances?state=failed'

But this does not provide me run_id for the instance. I would need the run_id to list the failed tasks of the DAG which I can use to decide the restart point of a DAG.
How do i get run_id for a task instance using airflow API

**Comment by potiuk** [comment]
I think it's missing - it's likely a bug when execution_date was replaced with dag_ru_id. Can you please open an issue for (or maybe even better make a PR for it?). I think in this case using execution_date might be the closest you can to use to do any kind of filtering (not perfect  but should work in most cases).
